src/stanc/stanc.ml

```diff
@@ -203,7 +198,12 @@ let use_file filename =
     if !print_model_cpp then print_endline cpp )
 
 let remove_dotstan s = String.drop_suffix s 5
-let model_name_check_regex = Str.regexp "^[a-zA-Z_].*$"
+
+let mangle =
+  String.concat_map ~f:(fun c ->
+      Char.(
+        if is_alphanum c || c = '_' then to_string c
+        else "x" ^ Int.to_string (to_int c)) )
```

Comment:
IIRC previously Stan converted `'-'` to `"_"`. It probably doesn't matter but we could keep it consistent.

Reply:
Explicitly changing - to _. Good call.